app/policies/stripe_card_policy.rb

```diff
@@ -14,7 +14,7 @@ def freeze?
   end
 
   def defrost?
-    if grantee? && !member? && (record.last_frozen_by != user)
+    if ((grantee? && !admin_or_manager?) || member_and_cardholder?) && (record.last_frozen_by != user)
       return false
     end
```

Comment:
```suggestion
    unless record.last_frozen_by == user || admin_or_manager?
        return false
    end
```
Thoughts on this code? I'm not sure if I'm missing an edge case but this feels a lot more readable.

Reply:
ik this got merged already but it lgtm. was busy for a few days.